doc/benchmarking.md

```diff
@@ -47,12 +46,30 @@ or using a regex filter to only run certain benchmarks.
 
 Notes
 ---------------------
-More benchmarks are needed for, in no particular order:
-- Script Validation
-- Coins database
-- Memory pool
-- Cuckoo Cache
-- P2P throughput
+
+Benchmarks help with monitoring for performance regressions and act as a scope
+for future performance improvements. They should cover components that impact
+performance critical functions of the system. Functions are performance
+critical, if their performance impacts users and the cost associated with a
+degradation in performance is high. A non-exhaustive list:
+
+- Initial block download (Cost: slow IBD results in full node operation being
```

Comment:
Users often complain about slow RPC calls - could we add those to this list?

Reply:
Could we clarify when we should use `benchmark::PriorityLevel::HIGH` and when the priority should be lower?

Reply:
I'd just like to avoid reviers claiming that they reject a PR because it's not in this list - but you're right, the "non-exhaustive list" should make that clear - you can resolve these